packages/block-editor/src/components/block-list/hovered-blocks.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * WordPress dependencies
+ */
+import { select, dispatch } from '@wordpress/data';
+
+/**
+ * Returns an array of block clientIds found under the cursor.
+ * Block clientIds are ordered from child to parent.
+ *
+ * @param {Object} event Mouse event object.
+ *
+ * @return {string[]} Array of block clientIds.
+ */
+export function getHoveredBlocksFromCursor( event ) {
+	const hoveredElements = document.elementsFromPoint(
+		event.clientX,
+		event.clientY
+	);
+	const blockIds = [];
+	hoveredElements.forEach( ( element ) => {
+		if ( element.dataset.block ) {
+			blockIds.push( element.dataset.block );
+		}
+	} );
+	return blockIds;
+}
+
+/**
+ * Conditionally evaluates and dispatches hovered block clientIds to the block-editor store.
+ *
+ * @param {Object} event Mouse event object.
+ */
+export function evaluateHoveredBlocks( event ) {
+	// Check last event data to determine if we should reevaluate hovered blocks.
+	// Requiring different mouseCoords limits unnecessary evaluations triggered from the same
+	// event firing from multiple nested blocks.  Checking the time allows us to re-evaluate
+	// even when mouseCoords have not changed, such as when scrolling via mouse-wheel.
+	// However, this deltaTime checked must be sufficiently long to ensure slow CPUs do not
+	// trigger unnecessary recalculations for events triggered simultaneously.
+	const { time, mouseCoords } =
```

Comment:
Scroll location would be much better to check than the last timeStamp to ensure this works on scrolling as well.  However, the scroll location corresponds to the `interface-interface-skeleteton__content` which is contained in the interface package.  Whether we found a way to reference this element or querySelector'd for it here, we would make this functionality dependent on a separate package which is also not ideal.  The time check will work, but fast mouse-wheel scrolls may not always re-trigger the evaluation.